core/src/main/java/org/elasticsearch/discovery/local/LocalDiscovery.java

```diff
@@ -82,9 +82,8 @@ public LocalDiscovery(Settings settings, ClusterName clusterName, ClusterService
         this.discoverySettings = new DiscoverySettings(settings, clusterSettings);
     }
 
-    @Override
-    public void setRoutingService(RoutingService routingService) {
-        this.routingService = routingService;
+    public void setAllocationService(AllocationService allocationService) {
```

Comment:
can you add the `@Override` back? (here and in all the other subclasses of `Discovery`)